cmd/presets/parser.go

```diff
@@ -91,56 +111,86 @@ func (p *DefaultParser) LookUpFiles(preset string) (foundFiles []string) {
 			continue
 		}
 
-		if _, err := osStat(fileName); !os.IsNotExist(err) {
+		if _, err := p.fs.Stat(fileName); !os.IsNotExist(err) {
 			foundFiles = append(foundFiles, fileName)
 		}
 	}
 	return
 }
 
-// WriteFiles write preset files
-func (p *DefaultParser) WriteFiles(preset string) (fileError string, err error) {
-	presetFiles := p.Presets[preset]
+// WriteFile write preset file
+func (p *DefaultParser) WriteFile(fileName string, fileContent string) (fileError string, err error) {
+	var (
+		file  afero.File
+		lines int
+	)
 
-	for fileName, fileContent := range presetFiles {
-		if strings.HasPrefix(fileName, "preset_") {
-			continue
-		}
+	file, err = p.fs.OpenFile(fileName, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, os.ModePerm)
 
-		var (
-			file  *os.File
-			lines int
-		)
-		file, err = os.OpenFile(fileName, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, os.ModePerm)
+	if err != nil {
+		fileError = fileName
+		return
+	}
 
-		if err != nil {
-			fileError = fileName
-			return
-		}
+	defer file.Close()
 
-		if lines, err = file.Write([]byte(fileContent)); err != nil {
-			fileError = fileName
-			return
-		}
+	if lines, err = file.Write([]byte(fileContent)); err != nil {
+		fileError = fileName
+		return
+	}
 
-		if len([]byte(fileContent)) != lines {
-			fileError = fileName
-			err = ErrPresetWriteAllBytes
-			return
-		}
+	if len([]byte(fileContent)) != lines {
+		fileError = fileName
+		err = ErrPresetWriteAllBytes
+		return
+	}
+
+	if err = file.Sync(); err != nil {
+		fileError = fileName
+		return
+	}
+
+	return
+}
+
+// GetPresetKeys get preset file contents
+func (p *DefaultParser) GetPresetKeys(preset string) (keys []string) {
+	presetData := p.Presets[preset]
+
+	for dataKey := range presetData {
+		keys = append(keys, dataKey)
+	}
 
-		if err = file.Sync(); err != nil {
-			fileError = fileName
+	sort.Strings(keys)
+
+	return
+}
+
+// GetPresetKeyContent get preset key value
+func (p *DefaultParser) GetPresetKeyContent(preset string, key string) (value string) {
+	presetData := p.Presets[preset]
+
+	for dataKey, dataContent := range presetData {
```

Comment:
presetData is a map right? better to do `dataContent , found = presetData[dataKey]` then.